spec.html

```diff
@@ -28152,14 +28467,107 @@ <h1>
+        <dl class="header">
+        </dl>
+
+        <emu-grammar>
+          WithClause : `with` `{` `}`
+        </emu-grammar>
+        <emu-alg>
+          1. Return a new empty List.
+        </emu-alg>
+
+        <emu-grammar>
+          WithClause : `with` `{` WithEntries `,`? `}`
+        </emu-grammar>
+        <emu-alg>
+          1. Let _attributes_ be WithClauseToAttributes of |WithEntries|.
+          1. Sort _attributes_ according to the lexicographic order of their [[Key]] fields, treating the value of each such field as a sequence of UTF-16 code unit values. NOTE: This sorting is observable only in that hosts are prohibited from changing behaviour based on the order in which attributes are enumerated.
+          1. Return _attributes_.
+        </emu-alg>
+
+        <emu-grammar>WithEntries : LiteralPropertyName `:` StringLiteral</emu-grammar>
+        <emu-alg>
+          1. Let _key_ be the PropName of |LiteralPropertyName|.
+          1. Let _entry_ be the ImportAttribute Record { [[Key]]: _key_, [[Value]]: SV of |StringLiteral| }.
+          1. Return « _entry_ ».
+        </emu-alg>
+
+        <emu-grammar>WithEntries : LiteralPropertyName `:` StringLiteral `,` WithEntries</emu-grammar>
+        <emu-alg>
+          1. Let _key_ be the PropName of |LiteralPropertyName|.
+          1. Let _entry_ be the ImportAttribute Record { [[Key]]: _key_, [[Value]]: SV of |StringLiteral| }.
+          1. Let _rest_ be WithClauseToAttributes of |WithEntries|.
+          1. Return the list-concatenation of « _entry_ » and _rest_.
+        </emu-alg>
+
+        <emu-note>
+          <p>This section is extended by <emu-xref href="#sec-deprecated-assert-keyword-for-import-attributes"></emu-xref>.</p>
+        </emu-note>
+      </emu-clause>
+
+      <emu-clause id="sec-deprecated-assert-keyword-for-import-attributes" deprecated normative-optional>
+        <h1>Deprecated `assert` keyword for Import Attributes</h1>
+
+        <p>Import Attributes are denoted using the `with` keyword. However, there are existing implementations based on a previous version of the proposal using the `assert` keyword. Due to potential web compatibility risks, the `assert` keyword is still included in this specification, but its removal is being investigated.</p>
+
+        <p>This deprecated feature includes the `assert` fallback for import attributes in import calls (step <emu-xref href="#step-evaluate-import-call-assert-fallback"></emu-xref> of <emu-xref href="#sec-evaluate-import-call"></emu-xref>).</p>
+
+        <p>The following augments the |ImportDeclaration| and |ExportDeclaration| productions in <emu-xref href="#sec-imports"></emu-xref> and <emu-xref href="#sec-exports"></emu-xref>:</p>
+        <!-- emu-grammar[type=definition] have different margin than other emu-grammar, but we want this to be aligned with the one below. -->
+        <emu-grammar style="display:block; margin-left: 5ex">
```

Comment:
I still think this should be marked `type="definition"` (and ecmarkup should be modified to treat it appropriately).

Reply:
(Actually, at this point, even if ecmarkup *isn't* modified, the effect should be pretty minor, since `AttributesKeyword` is the only affected nonterminal now.)

Reply:
I will try to modify ecmarkup. It already supports duplicated definitions, but only in annexes.